dlt/common/configuration/specs/base_configuration.py

```diff
@@ -169,7 +177,12 @@ def to_native_representation(self) -> Any:
     @classmethod
     def get_resolvable_fields(cls) -> Dict[str, type]:
         """Returns a mapping of fields to their type hints. Dunders should not be resolved and are not returned"""
-        return {f.name:f.type for f in cls.__dataclass_fields__.values() if not f.name.startswith("__")}
+        # Sort dynamic type hint fields last because they depend on other values
+        fields = sorted(
```

Comment:
please make sure that original order of fields is kept for the fields that do not have dynamic type hint. does `sorted()` preserve the order? are you sorting on a boolean variable?

Reply:
It is preserved. I'm sorting on bool  `key=lambda f: f.name in cls.__hint_resolvers__`  
So just the dynamic fields are resolved last. Otherwise original order is the same.   
It's mainly needed when dealing with a subclass like the fs destination config. Otherwise the hint for `credentials` gets resolved before `bucket_url` is known.

Reply:
https://docs.python.org/3/howto/sorting.html#sort-stability-and-complex-sorts  
> Sorts are guaranteed to be [stable](https://en.wikipedia.org/wiki/Sorting_algorithm#Stability). That means that when multiple records have the same key, their original order is preserved.

Reply:
amazing 👍